common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java

```diff
@@ -0,0 +1,67 @@
+package com.thekillerbunny.goofyplugin.lua;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.figuramc.figura.avatar.Avatar;
+import org.figuramc.figura.lua.FiguraLuaRuntime;
+import org.figuramc.figura.lua.LuaNotNil;
+import org.figuramc.figura.lua.LuaWhitelist;
+import org.figuramc.figura.lua.docs.LuaMethodDoc;
+import org.figuramc.figura.lua.docs.LuaMethodOverload;
+import org.figuramc.figura.lua.docs.LuaTypeDoc;
+
+@LuaWhitelist
+@LuaTypeDoc(name = "GoofyAPI", value = "goofy")
+public class GoofyAPI {
+    private final FiguraLuaRuntime runtime;
+    private final Avatar owner;
+
+    public GoofyAPI(FiguraLuaRuntime runtime) {
+        this.runtime = runtime;
+        this.owner = runtime.owner;
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class, String.class},
+                argumentNames = {"str", "pattern"}
+            )
+        },
+        value = "goofy.regex_match"
+    )
+    public String[] regexMatch(@LuaNotNil String str, @LuaNotNil String pattern) {
+        Pattern goofyPattern = Pattern.compile(pattern);
+        Matcher matcher = goofyPattern.matcher(str);
+        
+        ArrayList<String> matches = new ArrayList<String>();
+
+        while (matcher.find()) {
```

Comment:
a `gmatch`-like API would also be cool but I'll implement that tonight

Reply:
I agree, gmatch would be great. that's on you after this is merged